client/components/contact/ContactForm.jsx

```diff
@@ -0,0 +1,270 @@
+      clearFields();
+    };
+  }, [callShowErrorModal, callShowFeedbackSuccess, displayFeedbackSuccess, openErrorModal]);
+
+  // helper methods
+  function validateEmail(emailAddress) {
+    // eslint-disable-next-line
+    if (/^\w+([\.-]?\w+)*@\w+([\.-]?\w+)*(\.\w{2,3})+$/.test(emailAddress)) {
+      return true;
+    }
+    return false;
+  }
+
+  const clearErrors = useCallback(() => {
+    setFormValues(prevState => ({
+      ...prevState,
+      ...{
+        errors: {
+          missingFirstName: false,
+          missingLastName: false,
+          missingEmail: false,
+          invalidEmail: false,
+          missingMessage: false,
+        },
+      },
+    }));
+  }, []);
+
+  const validateForm = useCallback(() => {
+    const noFirstName = formValues.firstName.trim().length === 0;
+    const noLastName = formValues.lastName.trim().length === 0;
+    const noEmail = formValues.email.trim().length === 0;
+    const noMessage = formValues.message.trim().length === 0;
+    const incompleteEmail = (!noEmail && !validateEmail(formValues.email));
+    if (!noFirstName && !noLastName && !noEmail && !noMessage && !incompleteEmail) {
+      return true;
+    }
+
+    setFormValues(prevState => ({
+      ...prevState,
+      ...{
+        errors: {
+          missingFirstName: noFirstName,
+          missingLastName: noLastName,
+          missingEmail: noEmail,
+          invalidEmail: incompleteEmail,
+          missingMessage: noMessage,
+        },
+      },
+    }));
+    return false;
+  }, [formValues]);
+
+  // event handlers
+  const onInputChange = useCallback(event => {
+    event.preventDefault();
+    const { name, value } = event.target;
+    setFormValues(prevState => ({ ...prevState, [name]: value }));
+  }, []);
+
+  const handleSubmit = useCallback(event => {
+    event.preventDefault();
+
+    if (validateForm()) {
+      const body = [
+                `First name: ${formValues.firstName}`,
+                `Last name: ${formValues.lastName}`,
+                `Email: ${formValues.email}`,
+                `Association: ${formValues.association || 'Not provided'}`,
+                `Message: ${formValues.message}`,
+      ].join('\n');
+
+      setFormValues(prevState => ({
+        ...prevState,
+        ...{
+          loading: true,
+        },
+      }));
+
+      // dispatch action to redux with payload
+      callSendGitRequest({ title: formValues.email, body });
+    }
+  }, [callSendGitRequest,
+    formValues.association,
+    formValues.email,
+    formValues.firstName,
+    formValues.lastName,
+    formValues.message,
+    validateForm]);
+
+  return (
+    <Container maxWidth="sm">
+      <form id="contact-form" onSubmit={handleSubmit}>
+        <Grid container alignItems="center" justify="center" direction="column" style={{ gap: '10px' }}>
+          <Grid container alignItems="center" justify="center" direction="row" spacing={2}>
+            <Grid item xs={6}>
+              <TextField
```

Comment:
It looks like we have multiple TextFields with some subset of the same properties. Can we make a subclass or something of that sort so that we don't need to repeat all those properties?

Reply:
@jekijo or Jordan, any ideas?